tc/src/main/java/com/epam/tc/web/controller/CoursesController.java

```diff
@@ -15,11 +29,165 @@
+        Course course = getCourse(courseForm.getCourseId());
+        course.setName(courseForm.getName());
+        course.setDescription(courseForm.getDescription());
+        course.setLinks(courseForm.getLinks());
+        courseService.update(course);
+
+        resp.sendRedirect("/courses");
+    }
+
+    @RequestMapping(value = "/courses/{id}/subscribe", method = RequestMethod.GET)
+    public ModelAndView printForSubscribeCourse(@PathVariable(ID) String id) {
+        if (!"".equals(authenticatedUser.getUserEmail())) {
+            ModelAndView mav = new ModelAndView("subscribe");
+            mav.addObject("course", getCourse(id));
+            return mav.addObject("user", authenticatedUser.getUserEmail());
+        } else {
+            return new ModelAndView("403");
+        }
+    }
+
+    @RequestMapping(value = "/courses/{id}/subscribe", method = RequestMethod.POST)
+    public void SubscribeOnCourse(final HttpServletResponse resp,
+            @PathVariable(ID) String id) throws IOException {
+        courseService.addSubscriber(Integer.parseInt(id),
+                userService.getUserByEmail(authenticatedUser.getUserEmail()));
+        resp.sendRedirect("/courses");
+    }
+
+    @RequestMapping(value = "/courses/{id}/attend", method = RequestMethod.GET)
+    public ModelAndView printAttendess(@PathVariable(ID) String id) {
+        ModelAndView mav;
+        Course course = getCourse(id);
+        String username = authenticatedUser.getUserEmail();
+
+        if (course.isSubscribed(userService.getUserByEmail(username))) {
+            mav = new ModelAndView("attend");
+            mav.addObject("course", course);
+        } else {
+            mav = new ModelAndView("403");
+        }
+        return mav.addObject("user", authenticatedUser.getUserEmail());
+    }
+
+    @RequestMapping(value = "/courses/{id}/attend", method = RequestMethod.POST)
+    public void AttendOnCourse(final HttpServletResponse resp,
+            @PathVariable(ID) int courseId) throws IOException {
+        courseService.addAttender(courseId,
+                userService.getUserByEmail(authenticatedUser.getUserEmail()));
+        resp.sendRedirect("/courses");
+    }
+
+    @RequestMapping(value = "/courses/{id}/evaluate", method = RequestMethod.GET)
+    public ModelAndView evaluate(@PathVariable(ID) String id) {
+        ModelAndView mav;
+        Course course = getCourse(id);
+        User user = userService.getUserByEmail(authenticatedUser.getUserEmail());
+
+        if ((course.isAttended(user)) && (!course.isHaveGrade(user))) {
+            mav = new ModelAndView("evaluate");
+            mav.addObject("course", course);
+        } else {
+            mav = new ModelAndView("403");
+        }
+        return mav.addObject("user", authenticatedUser.getUserEmail());
+    }
+
+    private static final Logger LOG = LoggerFactory.getLogger(CoursesController.class);
+
+    @RequestMapping(value = "/courses/{id}/evaluate", method = RequestMethod.POST)
```

Comment:
Somewhere you use {id} as path variable name, somewhere {courseId} is used. It is better to use more meaningful one from this two.